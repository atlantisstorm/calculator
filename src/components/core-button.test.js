import React from 'react';
import { render, fireEvent } from "@testing-library/react";
import CoreButton from './core-button';

describe('CoreButton component', () => {
  it('should match snapshot', () => {
    const onClick = jest.fn();
    const value = "5";
    const cls = "";

    const { asFragment } = render(
      <CoreButton onClick={ onClick } value={ value } cls={ cls } />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should call dispatch with expected parameters', () => {
    const onClick = jest.fn();
    const value = "5";
    const cls = "";

    const { asFragment, getByTestId } = render(
      <CoreButton onClick={ onClick } value={ value } cls={ cls } />
    );

    const coreButton = getByTestId('core-button-5');
    fireEvent.click(coreButton);
    expect(onClick).toHaveBeenCalledTimes(1);

    // Plus loads of other tests...
  });
});