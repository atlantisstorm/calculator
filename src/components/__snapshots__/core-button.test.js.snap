// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoreButton component should match snapshot 1`] = `
<DocumentFragment>
  <button
    class=""
    data-testid="core-button-5"
    value="5"
  >
    5
  </button>
</DocumentFragment>
`;
